internal/service/federatedsettingsidentityprovider/data_source_federated_settings_identity_providers.go

```diff
@@ -257,6 +260,8 @@ func dataSourceMongoDBAtlasFederatedSettingsIdentityProvidersRead(ctx context.Co
 		return diag.FromErr(errors.New("federation_settings_id must be configured"))
 	}
 
+	// once the SDK is upgraded above version v20231115010 we can use pagination parameters to iterate over all results (and adjust documentation)
```

Comment:
knit: i'd remove this comment as we'll do this for all plural data sources

Reply:
Thought of leaving because for this case pagination parameters are not usable (in contrast to other data sources), but it is also true this wont be very urgent.